lib/vsc/filesystem/oceanstor.py

```diff
@@ -0,0 +1,148 @@
+from future.utils import with_metaclass
+
+import json
+import os
+import ssl
+
+from vsc.filesystem.posix import PosixOperations
+from vsc.utils import fancylogger
+from vsc.utils.patterns import Singleton
+from vsc.utils.rest import Client, RestClient
+from vsc.utils.py2vs3 import HTTPError, HTTPSHandler, build_opener
+
+OCEANSTOR_API_URL = 'https://172.19.96.130:8088/api/v2'
+
+
+class OceanStorClient(Client):
+    """Client for OceanStor REST API"""
+
+    def __init__(self, *args, ssl_verify=True, **kwargs):
+        """Wrapper for Client.__init__() allowing to disable SSL certificate verification"""
+        super(OceanStorClient, self).__init__(*args, **kwargs)
+
+        # X-Auth-Token header
+        self.x_auth_header = None
+
+        if ssl_verify is False:
+            # Disable verification of SSL certificates
+            nossl_context = ssl._create_unverified_context()
+            nosslHandler = HTTPSHandler(context=nossl_context)
+            self.opener = build_opener(nosslHandler)
+            fancylogger.getLogger().warning("Verification of SSL certificates disabled by request!")
+
+    def request(self, *args, **kwargs):
+        """
+        Wrapper for Client.request() with HTTP error and exit code handling
+        Injects X-Auth-Token headers into teh query if present
+        """
+        # Inject X-Auth-Token into headers (args=(method, url, body, headers))
+        args = list(args)
+        if args[3] is None:
+            args[3] = {}
+        if self.x_auth_header:
+            args[3].update(self.x_auth_header)
+
+        # Execute request catching any HTTPerror
+        try:
+            status, response = super(OceanStorClient, self).request(*args, **kwargs)
+        except HTTPError as err:
+            errmsg = "OceanStor query failed with HTTP error: %s (%s)" % (err.reason, err.code)
+            fancylogger.getLogger().error(errmsg)
+            raise
+
+        # Query exit code
+        try:
+            result = response['result']
+        except AttributeError as err:
+            errmsg = "OceanStor response lacks resulting status: %s" % response
+            fancylogger.getLogger().raiseException(errmsg, exception=AttributeError)
+        else:
+            ecmsg = "OceanStor query returned exit code: %s (%s)" % (result['code'], result['description'])
+            if result['code'] != 0:
+                fancylogger.getLogger().raiseException(ecmsg, exception=RuntimeError)
+            else:
+                fancylogger.getLogger().debug(ecmsg)
+
+        return status, response
+
+    def get_x_auth_token(self, password, username=None):
+        """Request authetication token"""
+        if not username:
+            username = self.username
+
+        query_url = os.path.join('aa', 'sessions')
+        payload = {
+            'user_name': username,
+            'password': password,
+        }
+
+        status, response = self.post(query_url, body=payload)
+        print(status, response)
```

Comment:
forgotten debug line?

Reply:
done in https://github.com/vub-hpc/vsc-filesystem-oceanstor/pull/1/commits/08a495f321ee43d6e510a25b7e46257b1a70bd67